### PR TITLE
Adding optional support for entity framework migrations.

### LINKED ISSUE #11: Exception thrown when adding fields to entity classes
**Describe the bug**
If you add a field to one of your entity classes it throws an exception when trying to read the cached db.

**To Reproduce**
1. Run your app such that it saves a sqlite db in the cache
2. Add a field to one of your entity classes.
3. Run the app again.
4. It attempts to select the new column from the cached sqlite db and since that new column doesn't exist it throws.

**Additional context**
Any workaround?

### LINKED ISSUE #23: Option to run `Database.MigrateAsync` instead of ` Database.CreateAsync`
https://github.com/JeremyLikness/SqliteWasmHelper/blob/main/SqliteWasmHelper/SqliteWasmDbContextFactory.cs#L80

image

We're relying on migrations in our project. Could you please accept option to run the followings instead:

```cs
await ctx.Database.MigrateAsync();
```

### PR DESCRIPTION
**Pull request by Eddie-Hartman**
Adds the ability to specify the migration strategy. An optional object "Migration" can be instantiated with a value of true and passed in when adding the context factory to allow entity framework migrations to be used.

Closes #11 
Closes #23

### CODE REVIEW

**Review by ysmoradi — APPROVED** [review]

**Comment by JeremyLikness** [comment]
Thanks everyone for your commitment and contributions. Deepy appreciated!

**Comment by Eddie-Hartman** [comment]
@JeremyLikness it seems like this still requires approval by a reviewer with write access. Would you mind approving or telling me what the next steps are?

**Comment by JeremyLikness** [comment]
I approved but it's its waiting for a build. Let me troubleshoot that.

**Comment by Eddie-Hartman** [comment]
Awesome. Thank you very much.

**Comment by ysmoradi** [comment]
Hi! Could you please publish the new nuget package so we can use these new enhancements?
Thanks in advance

**Comment by ysmoradi** [comment]
Ping!

**Comment by JeremyLikness** [comment]
Yes, will do this. Thanks for the pings!

**Comment by Eddie-Hartman** [comment]
@JeremyLikness ping

**Pull request merged**
